docs/subnets/customize-a-subnet.md

```diff
@@ -505,6 +507,122 @@ It's possible to enable this precompile with an initial configuration to activat
+    }
+  }
+}
+```
+
+`adminAddresses` denotes admin accounts who can add other `Admin` or `Enabled` accounts. `Admin` and `Enabled` are both eligible to change the current fee mechanism.
+
+The precompile implements the `RewardManager` interface which includes the `AllowList` interface. To see an example of the AllowList interface, see the [TxAllowList](#allowlist-interface) above.
+
+The `Stateful Precompile` contract powering the `RewardManager` adheres to the following Solidity interface at `0x0200000000000000000000000000000000000004` (you can load this interface and interact directly in Remix):
+
+```solidity
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+import "./IAllowList.sol";
+
+interface IRewardManager is IAllowList {
+  // setRewardAddress sets the reward address to the given address
+  function setRewardAddress(address addr) external;
+
+  // allowFeeRecipients allows block builders to claim fees
+  function allowFeeRecipients() external;
+
+  // disableRewards disables block rewards and starts burning fees
+  function disableRewards() external;
+
+  // currentRewardAddress returns the current reward address
+  function currentRewardAddress() external view returns (address rewardAddress);
+
+  // areFeeRecipientsAllowed returns true if fee recipients are allowed
+  function areFeeRecipientsAllowed() external view returns (bool isAllowed);
+}
+```
+
+RewardManager precompile uses `IAllowList` interface directly, meaning that it uses the same `AllowList` interface functions like `readAllowList` and `setAdmin`, `setEnabled`, `setNone`. For more information see [AllowList Solidity interface](#allowlist-interface).
+
+In addition to the AllowList interface, the RewardManager adds the following capabilities:
+
+- `setRewardAddress` - sets the address to which fees are sent. This address can be a contract or a user address. The address becomes the required coinbase address for the blocks that this mechanism is enabled on. Meaning that it will receive the fees collected from the transactions in the block. Receiving fees will not call any contract functions or fallback functions. It will simply send the fees to the address.
+
+- `allowFeeRecipients` - enables block producers to claim fees. This will allow block producers to claim fees by specifying their own addresses in their chain configs. See [here](#fee-recipient-1) for more information on how to specify the fee recipient address in the chain config.
+
+- `disableRewards` - disables block rewards and starts burning fees.
+
+- `currentRewardAddress` - returns the current reward address. This is the address to which fees are sent. It can include blackhole address (`0x010...0`) which means that fees are burned. It can also include a predefined hash (0x0000000000000000000000000000000000000000) denoting custom fee recipients are allowed. It's advised to use the `areFeeRecipientsAllowed` function to check if custom fee recipients are allowed first.
```

Comment:
```suggestion
- `currentRewardAddress` - returns the current reward address. This is the address to which fees are sent. It can include blackhole address (`0x010...0`) which means that fees are burned. It can also include a predefined hash (`0x0000000000000000000000000000000000000000`) denoting custom fee recipients are allowed. It's advised to use the `areFeeRecipientsAllowed` function to check if custom fee recipients are allowed first.
```